src/lib/starRelease.js

```diff
@@ -0,0 +1,86 @@
+import { linearSR, conditionalSR } from 'azimuth-js';
+import { Just, Nothing } from 'folktale/maybe';
+
+function mapRange(amount, cb) {
+  return [...Array(amount).keys()].map(cb);
+}
+
+// Preferentially withdraw from conditional contracts and use linear
+// to mop up the rest
+export function generateWithdrawTxs(contracts, batchLimits, amount, to) {
+  const [left, ...csrTxs] = generateConditionalWithdrawTxs(
+    contracts,
+    batchLimits,
+    amount,
+    to
+  );
+  const lsrTxs = generateLinearWithdrawTxs(contracts, left, to);
+  return [...csrTxs, ...lsrTxs];
+}
+
+function generateConditionalWithdrawTxs(
+  contracts,
+  batchLimits,
+  amount,
+  to,
+  offset = 0
+) {
+  // Successfully withdraw all requesed
+  if (amount <= 0) {
+    return [0];
+  }
+  // Return missing amount in car
```

Comment:
>car

?

Reply:
hangup from my lisping days, will change to head.